javalib/src/main/scala/java/lang/String.scala

```diff
@@ -1,42 +1,908 @@
-  def format(format: String, args: scala.Array[Object]): _String = ???
+  final val CASE_INSENSITIVE_ORDER: Comparator[_String] =
+    new CaseInsensitiveComparator()
+  private final val ascii = new Array[Char](128)
+  private var DefaultCharset: Charset = _
+  private var lastCharset: Charset    = _
+
+  for (i <- ascii.indices) {
+    ascii(i) = i.toChar
+  }
+
+  object ConsolePrintStream {
+    private var charset = "ISO-8859-1"
+//      AccessController.doPrivileged(new PriviActions[String]("console.encoding", "ISO8859_1"))
+
+    if (!Charset.isSupported(charset)) {
+      charset = "ISO-8859-1"
+    }
+  }
+
+  class ConsolePrintStream(out: java.io.OutputStream)
+      extends java.io.PrintStream(out, true) {
+    def print(str: _String) {
+      try {
+        if (str == null) {
+          write("null".getBytes(ConsolePrintStream.charset))
+        } else {
+          write(str.getBytes(ConsolePrintStream.charset))
+        }
+      } catch {
+        case e: java.io.IOException =>
+          setError()
+      }
+    }
+  }
+
+  private class CaseInsensitiveComparator
+      extends Comparator[_String] with Serializable {
+    def compare(o1: _String, o2: _String): Int =
+      o1.compareToIgnoreCase(o2)
+  }
+
+  def copyValueOf(data: Array[Char], start: Int, length: Int): _String =
+    new _String(data, start, length)
+
+  def copyValueOf(data: Array[Char]): _String =
+    new _String(data, 0, data.length)
+
+  def valueOf(data: Array[Char]): _String = new _String(data)
+
+  def valueOf(data: Array[Char], start: Int, length: Int): _String =
+    new _String(data, start, length)
+
+  def valueOf(value: Char): _String = {
+    val s =
+      if (value < 128) new _String(value, 1, ascii)
+      else new _String(0, 1, Array(value))
+    s.hashcode = value
+    s
+  }
+
+  def valueOf(value: scala.Double): _String = Double.toString(value)
+
+  def valueOf(value: scala.Float): _String = Float.toString(value)
+
+  def valueOf(value: scala.Int): _String = Integer.toString(value)
+
+  def valueOf(value: scala.Long): _String = Long.toString(value)
+
+  def valueOf(value: scala.Boolean): _String = if (value) "true" else "false"
+
+  def valueOf(value: AnyRef): _String =
+    if (value != null) valueOf(value.toString) else "null"
+
+  def format(_format: _String, args: AnyRef*): _String =
+    format(Locale.getDefault(), _format, args)
+
+  def format(loc: Locale, _format: _String, args: AnyRef*): _String = {
+    if (_format == null) {
+      throw new NullPointerException("null format argument")
+    } else {
+      val bufferSize =
+        if (args == null) _format.length + 0
+        else _format.length + args.length * 10
+      val f = new Formatter(new java.lang.StringBuilder(bufferSize), loc)
+      f.format(_format, args).toString
+    }
+  }
+
+  import scala.language.implicitConversions
+  private implicit def to_String(s: String): _String = valueOf(s.toCharArray)
+  private implicit def toString(s: _String): String  = s.toString()
```

Comment:
Without these implicit conversions, we can't compile this source.
It would be the cause of failure in build X(